third_party/pcsc-lite/naclport/server/src/readerfactory_nacl.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2016 Google Inc.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+// 3. The name of the author may not be used to endorse or promote products
+//    derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+// OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+// IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+// NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+// THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This file contains a replacement function for the original readerfactory.c
+// PC/SC-Lite internal implementation.
+
+// TODO: These includes need to come before readerfactory.h otherwise compiler
```

Comment:
I think that the readerfactory.h header from PC/SC-Lite is written a bit inaccurately, so that it actually requires including some other PC/SC-Lite headers before it.
Or you could also have problems if you tried to include C++ headers (like "ppapi/...") from inside "extern "C"", which won't work.

You can look up a working example here: https://github.com/GoogleChrome/chromeos_smart_card_connector/blob/master/third_party/pcsc-lite/naclport/server/src/google_smart_card_pcsc_lite_server/global.cc